include/ibamr/IBMethod.h

```diff
@@ -444,49 +444,69 @@ class IBMethod : public IBImplicitStrategy
      */
     void putToDatabase(SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> db) override;
 
-protected:
     /*!
      * Get the current structure position data.
+     *
+     * data_time must be equal to one of current time, new time, or half time. If this condition is met, X_data is set
```

Comment:
Are these time values available from `IBMethod`? If we are using them in this interface, it might be good to have accessors.

Reply:
I've add a `TimePoint` enum so that users can specify time points without knowing the actual time values.

Reply:
Nice. We should be doing this in `IBFEMethod` and other places too. 😁

Reply:
Everywhere else in the library we've made do with just a `double` to represent one of the three time values. In what context would we use a `TimePoint` instead of a `double` inside the library?

I'd like to avoid having two parallel interfaces which do the same things.

Reply:
Actually, most public interfaces to data do not rely on `double`'s. Eulerian data is retrieved via `getCurrentContext()` and `getNewContext()`. Data held by `IBFE` classes and `LDataManager` is retrieved through strings.

Another thing is that this change doesn't change the internals of the class. This only adds an additional public interface to the data owned by the class. The internal private interface stays the same.

> In what context would we use a `TimePoint` instead of a `double` inside the library?

This is useful for classes that don't know (or need to know) the time points used by `IBMethod`. My use case involves computing forces between structures time stepped by different `IBMethod` objects.